File: SchoolApi/Models/Course.cs
namespace SchoolApi.Models
{
    public class Course
    {
        public int ID { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int TeacherID { get; set; }
        public int? ClassroomID { get; set; }
        public int? TrainingID { get; set; }
        public int? SubModuleID { get; set; }
        public Teacher Teacher { get; set; }
        public Classroom Classroom { get; set; }
        public SubModule SubModule { get; set; }
        public Training Training { get; set; }
    }
}

File: SchoolSpa/src/app/_models/course.ts
import { Classroom } from "./classroom";
import { SubModule } from "./sub-module";
import { Teacher } from "./teacher";
import { Training } from "./training";

export interface Course{
    id : number,
    name : string,
    description : string,
    teacherID : number,
    classroomID : number,
    subModuleID : number,
    trainingID : number,
    teacher : Teacher,
    classroom : Classroom,
    subModule : SubModule,
    training : Training
}